### DISCUSSION TITLE
Using a global config parameters and rejits

### DISCUSSION THREAD

**Discussion by ASKabalan** [jax-ml/jax#24423] [Q&A] [answered]
Was wondering if there is an easy and minimal way to reproduce this behaviour with our own config parameters
```python
import jax
from functools import partial
from jax import numpy as jnp
@partial(jax.jit , static_argnums=(0,))
def f(shape):
    print(f"jitted for shape {shape}")
    return jnp.ones(shape)
jax.config.update("jax_enable_x64", False)
shape = (2 , 2 , 2)
f(shape)

# Out:
# jitted for shape (2, 2, 2) 
# Array([[[1., 1.],
#         [1., 1.]],
# 
#        [[1., 1.],
#         [1., 1.]]], dtype=float32)

jax.config.update("jax_enable_x64", True)

f(shape)
# Out:
# jitted for shape (2, 2, 2)
# Array([[[1., 1.],
#         [1., 1.]],
# 
#        [[1., 1.],
#         [1., 1.]]], dtype=float64)

```

Currently, if I change a global config parameter the function is not rejitted
As a workaround I use `jax.clear_caches()` but I am hoping for a cleaner solutions

Thank you

**Comment by jakevdp** [comment, marked as answer]
Hi - thanks for the question! The way that this works for configurations like `enable_x64` is via this function: https://github.com/jax-ml/jax/blob/f833891c87ed9e21bbae4bbda1fc5aa86fe5f328/jax/_src/config.py#L194-L201

This is an internal utility, and unfortunately there is no public API for customizing the trace context.

**Reply by ASKabalan** [reply]
Thank you for the quick answer @jakevdp !!

Can we clear a specific entry in the jit cache? without clearning everything
I can assume that it is also possible but with no public API aswell, no?

**Reply by jakevdp** [reply]
No, as far as I know there's no mechanism (public or non-public) for clearing a single cache entry.
